Sources/SWBApplePlatform/XCStringsCompiler.swift

```diff
@@ -189,7 +189,35 @@ public final class XCStringsCompilerSpec: GenericCompilerSpec, SpecIdentifierTyp
 
     /// Generates a task for compiling the .xcstrings to .strings/dict files.
     private func constructCatalogCompilationTask(_ cbc: CommandBuildContext, _ delegate: any TaskGenerationDelegate) async {
-        let commandLine = await commandLineFromTemplate(cbc, delegate, optionContext: discoveredCommandLineToolSpecInfo(cbc.producer, cbc.scope, delegate)).map(\.asString)
+        let isRunningInstallloc = cbc.scope.evaluate(BuiltinMacros.BUILD_COMPONENTS).contains("installLoc")
+
+        /// Custom lookup function to overwrite `XCSTRINGS_LANGUAGES_TO_COMPILE`
+        /// when `BUILD_ONLY_KNOWN_LOCALIZATIONS` is enabled for a regular build.
+        func lookup(_ macro: MacroDeclaration) -> MacroExpression? {
+            switch macro {
+            case BuiltinMacros.XCSTRINGS_LANGUAGES_TO_COMPILE:
+                guard !isRunningInstallloc else {
+                    // No need to intercept anything for installloc, its
+                    // language specification should always take precedence.
+                    return nil
+                }
+                if cbc.scope.evaluate(BuiltinMacros.BUILD_ONLY_KNOWN_LOCALIZATIONS), var knownLocalizations = cbc.producer.project?.knownLocalizations {
+
+                    knownLocalizations.removeAll(where: { $0 == "Base" })
+                    if !knownLocalizations.isEmpty {
+                        delegate.note("XCStrings will compile languages for known regions: \(knownLocalizations.joined(separator: ", "))")
```

Comment:
Just heads up that messages like this are going to show up for every xcstrings file, although in this particular case the specific file is not provided. Not saying it needs to be provided, but may appear as duplicate messages. Also for IB Compiler, etc.